python/paddle/distributed/fleet/meta_optimizers/amp_optimizer.py

```diff
@@ -58,7 +58,9 @@ def _init_wrapped_opt(self):
         # computation by split the check_finite_and_unscale op.
         is_distributed = self.role_maker._worker_num() > 1
         if self.user_defined_strategy.sharding:
+            # if self.user_defined_strategy.sharding or self.user_defined_strategy.model_parallel:
```

Comment:
I can't get it!

Reply:
sharding as well as  sharidng-megatron do not support the pure_fp16 allreduce logic introduced in amp&purefp16 metaoptimizer, so we need to set "is_distributed" = False while sharding or megatron enable.
This PR will add logic for sharding-megatron supporting, but megatron meta-optimizer will be add in another pr, so I will remove this line for now.